### DISCUSSION TITLE
Is there any way to programmatically set the Markdown content in a NestedLexicalEditor

### DISCUSSION THREAD

**Discussion by noteverseio** [mdx-editor/editor#353] [General]
Hey dear authors, thanks for the recent updates of this awesome editor, I have a new use case where I need to programmatically update or insert Markdown content in a NestedLexicalEditor, e.g. via Directive block or JSX block. E.g I need to write a Directive block where there is a button in the directive block and when customer clicks the button I will render editable markdown content retrieved from a remote server into the NestedLexicalEditor of the current Directive block.

Not sure if I can implement such features based on NestedLexicalEditor, any guidance would be really appreciated, thanks for your guidance.

**Comment by petyosi** [comment]
I'm not sure if this will completely solve your case, but you can use the `useMdastNodeUpdater` hook to get a function to update the node contents - https://mdxeditor.dev/editor/api/functions/useMdastNodeUpdater. This hook - https://mdxeditor.dev/editor/api/functions/useNestedEditorContext might also be helpful. 

Accessing the Lexical instance of the nested editor is not exposed at the moment, but I see how this can be useful. In general, the `activeEditor$` cell will let you get it if it is in focus, but this might not be what you need.
